src/Components/FeedbackPage/FeedbackForm/FeedbackForm.jsx

```diff
@@ -0,0 +1,204 @@
+import { useEffect, useState } from 'react';
+import { saveData } from '../../../DataService/localDataService';
+import { ErrorMessage } from '../../index';
+
+import styles from './FeedbackForm.module.scss';
+import { BiErrorCircle } from 'react-icons/bi';
+
+function FeedbackForm() {
+  const initialValues = {
+    username: '',
+    email: '',
+    comments: '',
+    newsletter: true,
+    feature: '',
+    feedbackType: ''
+  };
+  const [formData, setFormData] = useState(initialValues);
+  const [formErrors, setFormErrors] = useState({});
+  const [isSubmit, setIsSubmit] = useState(false);
+
+  function handleChange(event) {
+    const { name, value, type, checked } = event.target;
+    setFormData((prevFormData) => {
+      return {
+        ...prevFormData,
+        [name]: type === 'checkbox' ? checked : value
+      };
+    });
+  }
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    setFormErrors(validate(formData));
+    setIsSubmit(true);
+  }
+
+  function saveToLocalStorage() {
+    if (Object.keys(formErrors).length === 0 && isSubmit) {
+      saveData(formData);
+    }
+  }
+
+  useEffect(() => {
+    saveToLocalStorage();
+  }, [formErrors]);
+
+  function validate(values) {
+    const errors = {};
+    const emailRegEx =
+      /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+
+    if (!values.username) {
+      errors.username = '❌ Name is required!';
+    } else if (values.username.length < 3) {
+      errors.username = '❌ Must be at least 3 characters';
+    }
+    if (!values.email) {
+      errors.email = '❌ Email is required!';
+    } else if (!emailRegEx.test(values.email)) {
+      errors.email = '❌ This is not a valid email format!';
+    }
+    if (!values.comments) {
+      errors.comments = '❌ Please leave a comment!';
+    } else if (!values.comments.length < 20) {
+      errors.comments = '❌ Must be at least 20 characters';
+    }
+    if (!values.feature) {
+      errors.feature = '❌ Please select a feature!';
+    }
+    if (values.feedbackType === 'selection' || !values.feedbackType) {
+      errors.feedbackType = '❌ Please select feedback type!';
```

Comment:
i would recommend to move the cross icon into ErrorMessage component as far as it's not dynamic and should be shown for any error